src/boost_histogram/tag.py

```diff
@@ -112,26 +114,40 @@ def __call__(self, axis: AxisLike) -> int:  # noqa: ARG002
 
 class rebin:
     __slots__ = (
+        "axis",
+        "edges",
         "factor",
         "groups",
     )
 
     def __init__(
         self,
-        factor: int | None = None,
+        factor: int | PlottableAxis | None = None,
```

Comment:
If you want, you could make this a positional only optional arg, then assign to the correct one if it's given. Something like:

```python
self, factor_or_axis, /, factor=None, axis=None, ...)
```

Might make typing easier.